js/game.js

```diff
@@ -0,0 +1,94 @@
+/** @type {HTMLCanvasElement} */
+
+class Game {
+    constructor(ctx, width, height, player){
+        this.ctx = ctx;
+        this.width = width;
+        this.height = height;
+        this.player = player;
+        this.intervalId = null;
+        this.frames = 0;
+        this.enemies = [];
+        this.score = 0;
+    }
+
+    start() {
+        this.intervalId = setInterval(this.update, 1000 / 60);
+    }
+
+    update = () => {
+        this.frames++;
+        this.clear();
+        this.player.newPosition();
+        this.player.draw();
+        this.updateEnemies();
+        this.checkGameOver();
+        this.draw();
+        this.updateScore();
+    }
+
+    stop() {
+        clearInterval(this.intervalId);
+    }
+
+    clear() {
+        this.ctx.clearRect(0, 0, this.width, this.height);
+    }
```

Comment:
So we know that in every second we need to `clear` the `canvas` and `draw` everything again.

In the class example you used `clearRect` because there was no background just a white page.

But now we actually have a background that is the `road`, so instead of clearing the canvas with a white rectangle, we can use the road image to clear the canvas:

```
  clear() {
    let road = new Image();
    road.src = "../images/road.png";
    this.ctx.drawImage(road, 0, 0, this.width, this.height);
  }
```